scripts/update-metadata.mjs

```diff
@@ -0,0 +1,51 @@
+/* eslint-disable */
+import { execSync } from 'node:child_process';
+import { readFileSync, writeFileSync } from 'node:fs';
+import { dirname, join } from 'node:path';
+import { fileURLToPath } from 'node:url';
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = dirname(__filename);
+
+// Change directory to the root of the project
+process.chdir(join(__dirname, '..'));
+
+const backstageVersion = JSON.parse(
+  readFileSync('backstage.json', 'utf8'),
+).version.toString();
+
+export function updateBuildMetadata(backstageVersion) {
+  const buildMetadataPath = join(
+    process.cwd(),
+    'packages',
+    'app',
+    'src',
+    'build-metadata.json',
+  );
+  const buildMetadata = JSON.parse(readFileSync(buildMetadataPath, 'utf8'));
+
+  const rhdhVersion = JSON.parse(
+    readFileSync('package.json', 'utf8'),
+  ).version.toString();
+
+  // const commitTime = new Date().toISOString() is similar but returns millis too: 2024-05-03T12:12:08.174Z
+  // using the shell command allows us to have consistent datestamping everywhere, including sed+date transforms downstream
+  // unsure if this will break anyone developing on Windows; could switch to ISOString if it's a problem.
+  const commitTime = execSync('/usr/bin/date -u +%FT%TZ').toString().trim(); // eg., 2024-05-03T12:12:08Z
```

Comment:
@nickboldt I'm using macOS which doesn't have a `/usr/bin/date` file. Instead it's `/bin/date`.
```
> root@1.2.0 versions:metadata
> node ./scripts/update-metadata.mjs

Updating packages/app/src/build-metadata.json ...
/bin/sh: /usr/bin/date: No such file or directory
node:internal/errors:865
  const err = new Error(message);
              ^

Error: Command failed: /usr/bin/date -u +%FT%TZ
/bin/sh: /usr/bin/date: No such file or directory
```

Reply:
Blocked!!, I am also a Mac user, and I'm getting the same error and unable to make a commit. @nickboldt

Reply:
Workaround:

don't run this pre-commit hook:

```
$➔ cat .husky/pre-commit 
#!/usr/bin/env sh
. "$(dirname -- "$0")/_/husky.sh"

npm run lint-staged
npm run versions:metadata
```

instead run this:

```
$➔ cat .husky/pre-commit 
#!/usr/bin/env sh
. "$(dirname -- "$0")/_/husky.sh"

npm run lint-staged
# npm run versions:metadata
```

Reply:
this should work on all OSes. 

https://github.com/janus-idp/backstage-showcase/pull/1231/files